core/app/assets/javascripts/refinery/jquery.seourl.js

```diff
@@ -0,0 +1,282 @@
+/*
+ * jQuery SeoURL plugin 0.1
+ *
+ * http://www.spidersoft.com.au/projects/jquery-seo-url-plugin/
+ *
+ * Copyright (c) 2013 Slawomir Jasinski
+ *
+ * Dual licensed under the MIT and GPL licenses:
+ *   http://www.opensource.org/licenses/mit-license.php
+ *   http://www.gnu.org/licenses/gpl.html
+ *
+ */
+
+(function( $ ){
+
+  // protorype, and return string
+  String.prototype.seoURL = function (options) {
+
+    // Create some defaults, extending them with any options that were provided
+      var settings = $.extend( {
+        'transliterate': true,
+          'lowercase': false,
+          'uppercase': false,
+          'divider': '-',
+          'append': ''
+        }, options);
+
+
+    var text = this;
+
+    // transliterate
+    if (settings.transliterate === true) {
+      text = trans(text);
+    }
+    // lowercase
+    if (settings.lowercase === true) {
+      text = text.toLowerCase();
+    }
+    // uppercase
+    if (settings.uppercase === true) {
+      text = text.toUpperCase();
+    }
+
+    text = text.replace(/^\s+|\s+$/g, "") // trim leading and trailing spaces
+            .replace(/[_|\s]+/g, "-") // change all spaces and underscores to a hyphen
+            .replace(/[^a-zA-z\u0400-\u04FF0-9-]+/g, "") // remove almoust all characters except hyphen
+            .replace(/[-]+/g, "-") // replace multiple hyphens
+            .replace(/^-+|-+$/g, "") // trim leading and trailing hyphen
+            .replace(/[-]+/g, settings.divider) // replace hyphen with divider
+    return text + settings.append;
+  }
+
+
+  var text = '';
+
+  function trans (text) {
+    text = strtr(text, from, to)
+    text = strtr(text, special);
+    return text;
+  }
+
+  var from = 'ąĄęĘóÓśŚłŁżŻźŹćĆńŃ' + // polish
+         'ŠŒŽšœžŸ¥µÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÔÕÖØÙÚÛÜÝßàáâãäåæçèéêëìíîïðñòóôõöøùúûüýÿ'; // common
```

Comment:
:exclamation: missing some czech characters - č,ď,ň,ř,š,ť for example :exclamation:

I'm not sure if this is bulletproof library to generate slugs.

What about to use https://github.com/pid/speakingurl? I can create a gem for sprockets (like speakingurl-rails for example) to avoid adding this as vendored dependency since it looks like it is updating frequently.